public/app/plugins/panel/table/TableCellOptionEditor.tsx

```diff
@@ -0,0 +1,98 @@
+import { merge } from 'lodash';
+import React, { ReactNode, useState } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { TableCellOptions } from '@grafana/schema';
+import { Field, HorizontalGroup, Select, TableCellDisplayMode } from '@grafana/ui';
+
+import { BarGaugeCellOptionsEditor } from './cells/BarGaugeCellOptionsEditor';
+import { ColorBackgroundCellOptionsEditor } from './cells/ColorBackgroundCellOptionsEditor';
+
+const cellDisplayModeOptions = [
+  { value: TableCellDisplayMode.Auto, label: 'Auto' },
+  { value: TableCellDisplayMode.ColorText, label: 'Colored text' },
+  { value: TableCellDisplayMode.ColorBackground, label: 'Colored background' },
+  { value: TableCellDisplayMode.Gauge, label: 'Gauge' },
+  { value: TableCellDisplayMode.JSONView, label: 'JSON View' },
+  { value: TableCellDisplayMode.Image, label: 'Image' },
+];
+
+// Maps display modes to editor components
+interface ComponentMap {
+  [key: string]: Function;
```

Comment:
I really don't like this Function type here but couldn't figure out anything clever without spending hours how to type this properly. 😞

Reply:
Oh man, that's a bummer 😔.  I _think_ that `Function` is the preferred type for functional components, see https://github.com/facebook/create-react-app/pull/8177 and https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/ for some discussion on it

Reply:
I'm going to let this go but I'll try to find a solution for this in my free time because this just annoys me that I can't figure out :D